Extractor/Nodes/UAVariable.cs

```diff
@@ -516,14 +516,45 @@ public TimeSeriesCreate ToMinimalTimeseries(IUAClientAccess client, long? dataSe
             };
         }
 
-        public SourcedNodeWrite<CogniteExtractorTimeSeries> ToIdmTimeSeries(
+        public SourcedNodeWrite<CogniteExtractorTimeSeriesBase<JsonElement>> ToIdmTimeSeries(
             IUAClientAccess client,
             string space,
             string source,
             FullConfig config,
             Dictionary<string, string>? metaMap)
         {
-            var write = new CogniteExtractorTimeSeries
+            var writeAsJson = config.Cognite?.MetadataTargets?.Clean?.MetadataAsJson ?? false;
+
+            Dictionary<string, object> metaData;
+            if (writeAsJson)
+            {
+                metaData = BuildMetadataAsJson(config, client, true);
+            }
+            else
+            {
+                var stringMetadata = BuildMetadata(config, client, true);
+                metaData = stringMetadata.ToDictionary(kvp => kvp.Key, kvp => (object)kvp.Value);
+            }
+
+            Dictionary<string, JsonElement> extractedData;
+            try
+            {
+                var json = JsonSerializer.Serialize(metaData);
+                using var jsonDocument = JsonDocument.Parse(json);
+                var rootElement = jsonDocument.RootElement;
```

Comment:
I still don't love this, but I think you can avoid this whole issue if you use `JsonNode` directly instead.